tmt/base.py

```diff
@@ -463,6 +463,13 @@ def export(self, format_='yaml', keys=None):
 class Plan(Core):
     """ Plan object (L2 Metadata) """
 
+    extra_keys = [
```

Comment:
Naming this `_keys` instead of `extra_keys` will make it more consistent (see `Test` and `Story` classes) + the method `lint_keys` takes `_keys` attribute automatically so all you will need to do to lint is `invalid_keys = self.lint_keys(list(self.steps(...))`. Also, aren't you missing some attributes, such as `description`? It is part of the L0 metadata so it should also be supported in a plan, right? The rest of the PR looks good to me, please rebase to fix the failing tests.

Reply:
Hm, now that I am thinking about it a bit more, it now takes L0 into account as well... Because the `_keys` attribute is inehrited from class `Core`. But it is still a bit inconsistent with the rest.

@psss I'd like to know your opinion on this. Looking at the classes we have, we are basically "flattening" the keys right now and some things are being duplicated. For example `Test._keys` contains hardcoded `summary` even though it is part of L0 and not L1. I think that this could lead to problems if something is added to L0 metadata -- it won't be reflected in the `Test` class.

What if the `Test._keys` looked something like the following instead?

```
_keys = Core._keys + [
    'test',
    'path', ...]
```

This way the L0 keys are inherited from `Core` and are not duplicated and we just extend it with extra keys that come from L1/L2/L3. Does that make sense?

Reply:
I see, preventing the duplication by extending `Core._keys` would make sense. But there's one catch: The order of the keys is significant. It is used for example for presenting data in `Test.show()` which tries to organize the output into logical groups (e.g. `summary` first, `link` as the last). Do you see any easy way how to preserve a reasonable order while inheriting from the `Core` class?

Reply:
Hm, I see, didn't know that ordering was important. If we wanted to keep a reasonable order, we would definitely need to store it differently and it may become very complicated. One of my thoughts was storing the keys as a dict where keys are "sections" (logical groups)... But then you would also need to provide ordering... I don't see an easy way how to do this, can you find something better? If not, lets keep it simple (but not very elegant & future proof) and keep it as is + do the same for `Plan` class (basically taking everything from L0 and merging with L2 even though the keys will be repeated)

Reply:
Yeah, the only other solution which comes to my mind is to explicitly define the order directly under the `Test.show()` method. Or probably somewhere else because the `export()` uses it as well...